distributed/worker.py

```diff
@@ -2592,3 +2600,18 @@ def secede():
     duration = time() - thread_state.start_time
     worker.loop.add_callback(worker.maybe_transition_long_running,
                              thread_state.key, compute_duration=duration)
+
+
+class Reschedule(Exception):
+    """ Reschedule this task
+
+    Raising this exception will stop the current execution of the task and ask
+    the scheduler to reschedule this task, possibly on a different machine.
+
+    This does not guarantee that the task will move onto a different machine.
+    The scheduler will proceed through its normal heuristics to determine the
+    optimal machine to accept this task.  The machine will likely change if the
+    load across the cluster has significantly changed since first scheduling
+    the task.
+    """
```

Comment:
How does this interact with the scheduler's tracking of suspicious tasks? I assume a task that raises a Reschedule still counts as a failure, but it'd be good to document that here.

Reply:
It does not count as a failure.  A task that raises Reschedule will not be considered more suspicious.